AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_DiffuseProbeGridAdded.py

```diff
@@ -0,0 +1,187 @@
+        "Diffuse Probe Grid component enabled",
+        "Diffuse Probe Grid component was not enabled")
+    enter_game_mode = (
+        "Entered game mode",
+        "Failed to enter game mode")
+    exit_game_mode = (
+        "Exited game mode",
+        "Couldn't exit game mode")
+    is_visible = (
+        "Entity is visible",
+        "Entity was not visible")
+    is_hidden = (
+        "Entity is hidden",
+        "Entity was not hidden")
+    entity_deleted = (
+        "Entity deleted",
+        "Entity was not deleted")
+    deletion_undo = (
+        "UNDO deletion success",
+        "UNDO deletion failed")
+    deletion_redo = (
+        "REDO deletion success",
+        "REDO deletion failed")
+
+
+def AtomEditorComponents_DiffuseProbeGrid_AddedToEntity():
+    """
+    Summary:
+    Tests the Diffuse Probe Grid component can be added to an entity and has the expected functionality.
+
+    Test setup:
+    - Wait for Editor idle loop.
+    - Open the "Base" level.
+
+    Expected Behavior:
+    The component can be added, used in game mode, hidden/shown, deleted, and has accurate required components.
+    Creation and deletion undo/redo should also work.
+
+    Test Steps:
+    1) Create a Diffuse Probe Grid entity with no components.
+    2) Add a Diffuse Probe Grid component to Diffuse Probe Grid entity.
+    3) UNDO the entity creation and component addition.
+    4) REDO the entity creation and component addition.
+    5) Verify Diffuse Probe Grid component not enabled.
+    6) Add Shape component since it is required by the Diffuse Probe Grid component.
+    7) Verify Diffuse Probe Grid component is enabled.
+    8) Enter/Exit game mode.
+    9) Test IsHidden.
+    10) Test IsVisible.
+    11) Delete Diffuse Probe Grid entity.
+    12) UNDO deletion.
+    13) REDO deletion.
+    14) Look for errors.
+
+    :return: None
+    """
+
+    import azlmbr.legacy.general as general
+    import azlmbr.render as render
+
+    from editor_python_test_tools.editor_entity_utils import EditorEntity
+    from editor_python_test_tools.utils import Report, Tracer, TestHelper
+    from Atom.atom_utils.atom_constants import AtomComponentProperties
+
+    with Tracer() as error_tracer:
+        # Test setup begins.
+        # Setup: Wait for Editor idle loop before executing Python hydra scripts then open "Base" level.
+        TestHelper.init_idle()
+        TestHelper.open_level("", "Base")
+
+        # Test steps begin.
+        # 1. Create a Diffuse Probe Grid entity with no components.
+        diffuse_probe_grid_entity = EditorEntity.create_editor_entity(AtomComponentProperties.diffuse_probe_grid())
+        Report.critical_result(Tests.diffuse_probe_grid_creation, diffuse_probe_grid_entity.exists())
+
+        # 2. Add a Diffuse Probe Grid component to Diffuse Probe Grid entity.
+        diffuse_probe_grid_component = diffuse_probe_grid_entity.add_component(AtomComponentProperties.diffuse_probe_grid())
```

Comment:
while we don't follow specific line lengths, this one is still a bit too long, pyCharm defaults to 120 characters but PEP-8 suggests 79 characters (this is due to different monitor resolutions and how much code they can fit on the screen at one time)
we can stick to 120 but that would still make this too long, recommend formatting like this to reduce line length:
```
diffuse_probe_grid_component = diffuse_probe_grid_entity.add_component(
AtomComponentProperties.diffuse_probe_grid())
```